packages/legend-query-builder/src/components/QueryBuilderConstantExpressionPanel.tsx

```diff
@@ -43,45 +42,80 @@ import {
   Multiplicity,
   isValidIdentifier,
 } from '@finos/legend-graph';
-import { generateEnumerableNameFromToken } from '@finos/legend-shared';
+import {
+  deepClone,
+  generateEnumerableNameFromToken,
+} from '@finos/legend-shared';
 import { observer } from 'mobx-react-lite';
 import { DEFAULT_CONSTANT_VARIABLE_NAME } from '../stores/QueryBuilderConfig.js';
 import type { QueryBuilderState } from '../stores/QueryBuilderState.js';
 import {
   type QueryBuilderConstantExpressionState,
   QueryBuilderSimpleConstantExpressionState,
   QueryBuilderCalculatedConstantExpressionState,
+  cloneQueryBuilderConstantLambdaEditorState,
 } from '../stores/QueryBuilderConstantsState.js';
 import { buildDefaultInstanceValue } from '../stores/shared/ValueSpecificationEditorHelper.js';
 import { BasicValueSpecificationEditor } from './shared/BasicValueSpecificationEditor.js';
 import { QUERY_BUILDER_TEST_ID } from '../__lib__/QueryBuilderTesting.js';
 import { QUERY_BUILDER_DOCUMENTATION_KEY } from '../__lib__/QueryBuilderDocumentation.js';
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { variableExpression_setName } from '../stores/shared/ValueSpecificationModifierHelper.js';
 import { LambdaEditor } from './shared/LambdaEditor.js';
 import { VariableViewer } from './shared/QueryBuilderVariableSelector.js';
 import { flowResult } from 'mobx';
 
+const getConstantNameValidationMessage = (
+  constantInput: string,
+  allVariableNames: string[],
+  stateName: string,
```

Comment:
what do we mean by stateName here? maybe a better name would explain the funtion a bit better.

Reply:
It's the current name of the constant in the state. I'll rename it to be clearer.